pgcli/main.py

```diff
@@ -405,6 +420,19 @@ def get_completions(self, text, cursor_positition):
         return self.completer.get_completions(
             Document(text=text, cursor_position=cursor_positition), None)
 
+    def toggle_expanded_output(self):
+        """Callback for the \\x special command"""
+        self.use_expanded_output = not self.use_expanded_output
+        message = u"Expanded display is "
+        message += u"on." if self.use_expanded_output else u"off."
+        return [(None, None, None, message)]
+
+    def toggle_timing(self):
+        self.timing_enabled = not self.timing_enabled
+        message = "Timing is "
+        message += "on." if self.timing_enabled else "off."
+        return [(None, None, None, message)]
+
```

Comment:
What PGSpecial provides is a central place to keep the state about the various special commands. Even now PGSpecial's timing and expanded are just states that are reported to PGCli and we choose to act based on that state.

I think it is reasonable to provide a place to maintain state that is related to the special commands.